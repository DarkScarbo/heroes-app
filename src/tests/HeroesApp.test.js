import React from 'react';
import { mount } from 'enzyme';
import { AuthContext } from '../auth/AuthContext';
import { HeroesApp } from '../HeroesApp';

describe('Tests in HeroesApp', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Xabi'
        }
    };

    test('should mount the component correctly', () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                    <HeroesApp />
            </AuthContext.Provider>
        );
        
        expect( wrapper ).toMatchSnapshot();
    })
    
})
