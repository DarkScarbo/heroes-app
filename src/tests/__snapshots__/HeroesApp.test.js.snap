// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in HeroesApp should mount the component correctly 1`] = `
<div>
  <div
    className="card text-center ms-5 animate__animated animate__fadeIn text-white bg-dark"
  >
    <div
      className="card-body"
    >
      <form
        className="container mt-3"
        onSubmit={[Function]}
      >
        <div
          className="input-group mb-3"
        >
          <input
            autoComplete="off"
            autoFocus={true}
            className="form-control"
            name="name"
            onChange={[Function]}
            placeholder="Your name"
            required={true}
            type="text"
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-primary"
              type="submit"
            >
              Login!
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
